budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
         self.assertEqual(
             other_parties, ["Service Provider: DOPW\nService Provider: AAAA"]
         )
+
+
+class ProvInfraProjectAPITestCase(APITestCase):
+    def setUp(self):
+        """Create 30 Provincial Infrastructure Projects"""
+
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.url = reverse("search-provincial-infrastructure-projects")
+        self.province = u"Eastern Cape"
+        self.status = u"Construction"
+        self.source = u"Community Library Service Grant"
+        self.department = u"Department 1"
```

Comment:
please move these things that are only used in the test cases to those test cases

the most important thing about a test case is that it's easy to verify what it's testing, and that it's doing that correctly.

To be able to do that, you want data locality - you don't want to have to look in several places to understand what the test case is doing - as much as possible of the test should be inside the test case. I'll make a note in a test case to explain further

Reply:
they're moved into related tests.

https://github.com/vulekamali/datamanager/pull/190/commits/ee17bc29b03185e791fa6fa2001619b1076a4683